src/masternode-sync.cpp

```diff
@@ -53,17 +53,24 @@ std::string CMasternodeSync::GetAssetName()
 
 void CMasternodeSync::SwitchToNextAsset(CConnman& connman)
 {
+    CDeterministicMNList mnList;
     switch(nCurrentAsset)
     {
         case(MASTERNODE_SYNC_FAILED):
             throw std::runtime_error("Can't switch to next asset from failed, should use Reset() first!");
             break;
         case(MASTERNODE_SYNC_INITIAL):
+            // try to update UI with the current MN list/count (can fail since we might be still syncing)
+            mnList = deterministicMNManager->GetListAtChainTip();
+            uiInterface.NotifyMasternodeListChanged(mnList);
```

Comment:
Why is this needed here? I assume because the wallet is otherwise starting up and NotifyMasternodeListChanged never gets called until the list actually changes?

If that's what this is for, then may it's better to look into how/when `CDSNotificationInterface::InitializeCurrentBlockTip` is invoked and do it the same way, e.g. call `uiInterface.NotifyMasternodeListChanged` directly after `InitializeCurrentBlockTip`.